README.md

```diff
@@ -47,81 +73,376 @@ The following sections specify the required and optional content for `eaw.modinf
+
+**Level:** **REQUIRED**
+
+**Data Type**: `String`
+
+**Data Semantics**: File System Path
+
+**Description:**
 
 The content folder's name as specified by the Steam Uploader.
 
-#### The `"steamdata.visibility"` Tag
+### The `"steamdata.visibility"` Property
+
+**Level:** **REQUIRED**
+
+**Data Type**: `integer`
+
+**Data Semantics**: Enum
+
+**Description:**
 
 The visibility enumeration:
 
 | Value | Meaning |
 |:--:|:--|
-|0|hidden|
-|1|friends only|
-|2|public|
+|`0`|hidden|
+|`1`|friends only|
+|`2`|public|
+
+
+### The `"steamdata.metadata"` Property
+
+**Level:** **REQUIRED**
+
+**Data Type**: `String`
 
+**Data Semantics**: Text
 
-#### The `"steamdata.metadata"` Tag
+**Description:**
 
 Arbitrary metadata as string.
 
-#### The `"steamdata.tags"` Tag
+### The `"steamdata.tags"` Property
+
+**Level:** **REQUIRED**
+
+**Data Type**: `String[]`
+
+**Data Semantics**: Collection of tags
+
+**Description:**
 
 Steam Tags as specified by the Steam Uploader.
 
 At least either `EAW` or `FOC` is required to determine the game the mod shows up for.
 
-### The `"custom"` Tag [OPTIONAL]
+## Dependency Resolving
+The game supports chaining (could also be called *overriding* or *linked*) mods by queuing up the command line arguments `STEAMMOD` and/or `MODPATH`. 
+While the Command Line options only can resemble a line (or more precisely a Queue), real mod dependencies can look like complex tree due to multiple inheritance. Thus there needs to be a deterministic logic to convert the real hierachy into a CLI compatible representation. 
+
+To describe relations between mods this specification introduces an optional [`dependency`](#the-dependencies-property) property which is an ordered list. Each item in the list branches from current mod, which fullfils the multiple inheritance logic.
+
+The conversion from a tree structure to a line is called *flattening*.
+Implementations of this standard must fullfill the following requirements to create a sucessfull conversion:
+
+1. The `dependencies` list MUST be processed from start to end.
+2. Before processing dependiencies of `n+1` (where n is the current processed mod) all dependencies of `n` must be processed first.
+3. Duplicates MUST get removed. *Having duplicates that are not direct neighbours is a strong indicator you have a dependency cycle!*
+4. The implementation MUST be able to recognice and react on dependency cycles. 
+5. The provided [test cases](#dependency-resolving-test-cases) MUST pass:
+6. The result of the *flattening* MAY get reversed depending on the requirements of the consumer.
+
+*Note: Converting a (non-binary) tree structure to a flatteded and duplicate-free line is a one-way operation. It also looses accuracy. The requirements above are created to ensure consistency across multiple implementations.*
+
+*Advise: For making sure you mod always work as intended, we recommend to keep the inheritance level to a minimum. Also try to avoid multiple inheritances.*
```

Comment:
```suggestion
*Advise: For making sure your mod always works as intended, we recommend to keep the inheritance level to a minimum. Also try to avoid multiple inheritances.*
```